fixtures/nesting/src/legacy/createLegacyRoot.js

```diff
@@ -0,0 +1,37 @@
+/* eslint-disable react/jsx-pascal-case */
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+import ThemeContext from './shared/ThemeContext';
+
+// Note: this is a semi-private API, but it's ok to use it
+// if we never inspect the values, and only pass them through.
+import {__RouterContext} from 'react-router';
+
+// Pass through every context required by this tree.
+// The context object is populated in src/modern/withLegacyRoot.
+function Bridge({children, context}) {
+  return (
+    <ThemeContext.Provider value={context.theme}>
+      <__RouterContext.Provider value={context.router}>
+        {children}
+      </__RouterContext.Provider>
+    </ThemeContext.Provider>
+  );
+}
+
+export default function createLegacyRoot(container) {
```

Comment:
This intentionally mirrors `createRoot` so it's easier to transition a file like this to it when it exists — for people bridging in the reverse direction.